Automattic/wp-calypso
# ActivityLog: Add support for posts nested filters

Resolves https://github.com/Automattic/jetpack-backup-team/issues/216

⚠️ This PR depends on D127934-code being applied in your WPCOM sandbox. That PR enables the nested filter support in WPCOM API.

## Proposed Changes

* Add support for Activity Type nested filters UI
  * When selecting a parent filter, it should autoselect all children filters.
  * When unselecting a parent filter, it should autoselect all children filters.
  * ìndeterminate` status for parent filter checkboxes is not yet implemented. It will be addressed in another PR as it requires refactoring to use [CheckboxControl](https://wordpress.github.io/gutenberg/?path=/docs/components-checkboxcontrol--docs) instead of FormCheckbox.

| Before | After |
|---|---|
| ![CleanShot 2023-11-08 at 18 06 58@2x](https://github.com/Automattic/wp-calypso/assets/1488641/94db5b91-9d0f-490d-8eef-c93d1be5c3b5) | ![CleanShot 2023-11-08 at 18 08 00@2x](https://github.com/Automattic/wp-calypso/assets/1488641/3169db25-9992-4cc8-922a-f735cc1aed6f) |

### Demo
![CleanShot 2023-11-08 at 18 09 26](https://github.com/Automattic/wp-calypso/assets/1488641/605a1b21-27ab-406a-9525-5a80689c92e3)

## Testing Instructions
* Apply D127934-code in your sandbox.
* Spin up a Calypso and Jetpack Cloud live branch
* Navigate to the Activity Log with a site that has recent posts published, updated, deleted and/or trashed.
* On the `Activity Type` filter, ensure that you see `Posts and Pages` and indented, another filters like `Published`, `Trashed`, `Deleted` and/or `Updated`.
* Play around with the filters and ensure it filters the results as expected.
* Ensure clicking on `Posts and Pages` check (or uncheck) all children filters.

## Pre-merge Checklist
